dhis-2/dhis-services/dhis-service-validation/src/main/java/org/hisp/dhis/validation/outlierdetection/ValidationOutlierDetectionRequest.java

```diff
@@ -0,0 +1,122 @@
+ * modification, are permitted provided that the following conditions are met:
+ * Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ *
+ * Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ * Neither the name of the HISP project nor the names of its contributors may
+ * be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.validation.outlierdetection;
+
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.hisp.dhis.common.IllegalQueryException;
+import org.hisp.dhis.feedback.ErrorCode;
+import org.hisp.dhis.feedback.ErrorMessage;
+import org.hisp.dhis.outlierdetection.OutlierDetectionAlgorithm;
+import org.hisp.dhis.outlierdetection.OutlierDetectionRequest;
+import org.hisp.dhis.setting.SettingKey;
+import org.hisp.dhis.setting.SystemSettingManager;
+import org.springframework.stereotype.Component;
+
+@Component
+@AllArgsConstructor
+@Slf4j
+public class ValidationOutlierDetectionRequest {
+
+  private final SystemSettingManager systemSettingManager;
+
+  /**
+   * Validates the given request.
+   *
+   * @param request the {@link OutlierDetectionRequest}.
+   * @throws IllegalQueryException if request is invalid.
+   */
+  public void validate(OutlierDetectionRequest request, boolean isAnalytics)
+      throws IllegalQueryException {
+    ErrorMessage errorMessage = validateForErrorMessage(request, isAnalytics);
+
+    if (errorMessage != null) {
+      log.warn(
+          String.format(
+              "Outlier detection request validation failed, code: '%s', message: '%s'",
+              errorMessage.getErrorCode(), errorMessage.getMessage()));
+
+      throw new IllegalQueryException(errorMessage);
+    }
+  }
+
+  private ErrorMessage validateForErrorMessage(
+      OutlierDetectionRequest request, boolean isAnalytics) {
+
+    int maxLimit =
```

Comment:
We had agreed to leave it hard-coded to 500 as the limit. I believe this can be removed.
The injection of `SystemSettingManager systemSettingManager` will not be needed, consequently.

Reply:
? we agreed that the old implementation will be not changed (500 hard coded, for performance sake). The new analytics implementation should respect system settings...